pkg/network/tracer/gateway_lookup.go

```diff
@@ -39,10 +39,16 @@ func newGatewayLookup(config *config.Config, runtimeCompilerEnabled bool, m *man
 	}
 
 	var router network.Router
+	var err error
 	if runtimeCompilerEnabled {
-		router = newEbpfRouter(m)
+		router, err = newEbpfRouter(m)
 	} else {
-		router = network.NewNetlinkRouter(config.ProcRoot)
+		router, err = network.NewNetlinkRouter(config.ProcRoot)
+	}
+
+	if err != nil {
```

Comment:
Can you also check for `router == nil` here? If the `GetMap` call returns `ok = false, err = nil`, then this case will happen.